geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/protobufs/TileProtoBuf.scala

```diff
@@ -126,13 +126,31 @@ trait TileProtoBuf {
     }
 
     def decode(message: ProtoTile): Tile = {
-      val ct = messageToCellType(message.cellType.get)
+      val messageCellType = message.cellType.get
+      val ct = messageToCellType(messageCellType)
 
       message.cellType.get.dataType.toString match {
-        case ("INT" | "BYTE" | "SHORT") =>
-          ArrayTile(message.sint32Cells.toArray, message.cols, message.rows).interpretAs(ct)
-        case ("UBYTE" | "USHORT" | "BIT") =>
-          ArrayTile(message.uint32Cells.toArray, message.cols, message.rows).interpretAs(ct)
+        case "BIT" =>
+            ArrayTile(message.uint32Cells.toArray, message.cols, message.rows).interpretAs(ct)
+
+        case ("BYTE" | "SHORT") =>
+          if (messageCellType.hasNoData)
+            ArrayTile(message.sint32Cells.toArray, message.cols, message.rows)
+              .interpretAs(ct)
+              .map { x => if (x == Int.MinValue) messageCellType.nd.toInt else x }
```

Comment:
Ah, ArrayTile comes in with constant no data value. If you use RawArrayTile it will come in with no nodata value and you can just use interpretAs, should not even need the if statement